pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go

```diff
@@ -491,9 +491,16 @@ func (d *Device) Interface() string {
 	return d.DeviceInterface
 }
 
-// CIDR implements the MachineNetwork interface.
-func (d *Device) CIDR() string {
-	return d.DeviceCIDR
+// Addresses implements the MachineNetwork interface.
+func (d *Device) Addresses() []string {
+	switch {
+	case len(d.DeviceAddresses) > 0:
+		return append([]string(nil), d.DeviceAddresses...)
+	case d.DeviceCIDR != "":
```

Comment:
Should we accept both cases, where someone fills both `addresses:` _and_ `cidr:`?

Reply:
Meh.  Probably unnecessary complexity.

Reply:
this is considered invalid in the validation code, so we consider that to be an error. I think it's less confusing to say that one should use one or another, but not both